pytorch_toolkit/object_detection/tests/train_tests_custom_object_detection.py

```diff
@@ -68,29 +68,12 @@ def setUp(self):
             self.output_folder = os.path.join(self.template_folder, f'output_{self.id()}')
             os.makedirs(self.output_folder, exist_ok=True)
 
-        def do_evaluation(self, on_gpu):
-            initial_command = 'export CUDA_VISIBLE_DEVICES=;' if not on_gpu else ''
-            metrics_path = os.path.join(self.output_folder, "metrics.yaml")
-            run_through_shell(
-                f'{initial_command}'
-                f'cd {self.template_folder};'
-                f'python eval.py'
-                f' --test-ann-files {self.ann_file}'
-                f' --test-data-roots {self.img_root}'
-                f' --save-metrics-to {metrics_path}'
-                f' --load-weights {os.path.join(self.output_folder, "latest.pth")}'
-            )
-
-            with open(metrics_path) as read_file:
-                content = yaml.safe_load(read_file)
-
-            for metric_key in metric_keys:
-                value = [metrics['value'] for metrics in content['metrics'] if metrics['key'] == metric_key][0]
-                self.assertGreaterEqual(value, 0.0)
-
-        def do_finetuning(self, on_gpu):
+        def do_finetuning(self, with_classes, on_gpu):
             log_file = os.path.join(self.output_folder, 'test_finetuning.log')
             initial_command = 'export CUDA_VISIBLE_DEVICES=;' if not on_gpu else ''
+            classes = ''
+            if with_classes:
+                classes = 'person,vehicle,non-vehicle'
```

Comment:
Ilya, why do you test only cases "all classes" and "3 classes"?
As far as I remember, the test uses airport dataset that contains 3 classes -- is it correct?
If yes, both of the use cases check similar functionality.

Could you make several tests with different number of classes? (`num_classes` from 1 to 3 with special value `None`) 

But I think that in this case the test itself should not change the original folder where the model template is instantiated.
Instead of that each of the tests should make a copy of the model template folder (with special name `= original_name + test.id()`) and run tests inside -- I made it for NNCF tests, see `create_object_detection_nncf_test_case`
(Note that for this approach each tests class should have its own different number of classes, whereas now each test function has its number of classes)

Reply:
what is `num_classes`?

what are you talking about? please don't mix questions/concerns in a single message.
not enough?

def setUp(self):
            self.output_folder = os.path.join(self.template_folder, f'output_{self.id()}')
            os.makedirs(self.output_folder, exist_ok=True)

Reply:
Ilya, it is one question/concern.
I think you should make several tests (trest classes) for different number of classes.
But to implement it you will have to make copying of instantiated template folder to new folders, like I do in NNCF tests...

Reply:
Ok, after consideration -- there is no need to copy the whole model template folder in the test

Reply:
added more variants of classes

Reply:
Ilya, I re-checked: yes, you are right, the file `data/airport/annotation_example_train.json` contains only one class `person`.
But I found the annotation file `data/airport/annotation_faces_train.json` -- it contains faces! (I re-checked that -- drawed the faces, they are correct)
I think these two annotation files may be merges: we can use as the base the file `annotation_example_train.json`  and merge into it `data/airport/annotation_faces_train.json` 
I sent you my script to merge them and the result of merging

So, now we can use in tests two classes (faces and persons) and one class (persons)

Reply:
I think its enough to test what is presented now. You can modify test data by yourself if you really want.